### PR TITLE
Bug 2100857: Fix master's manifest file pattern for oVirt provider

### PR DESCRIPTION
**Pull request by mmartinv**
It looks like the master's manifest file names can change between
openshift installer versions and makes the assisted installer to fail
when activating the oVirt provider.

The proposed changes make use of file globs to find the actual
master files.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [] No tests needed

## Assignees

/cc @eliorerz 
/cc @osherdp 
/cc @michaellevy101 

## Checklist

- [x] Title and description added to both, commit and PR.
- [] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by osherdp — CHANGES_REQUESTED** [review]
Looks good, just add unit tests when done

**Comment by mmartinv** [comment]
> Looks good, just add unit tests when done

Done, please take a look as I needed to make some more changes.
Once you're done could you please take a look at  #3972 which depends on this PR?

File: internal/provider/ovirt/ignition.go [hunk 1]
```diff
@@ -77,43 +80,87 @@ func getOvirtClient(params *models.OvirtPlatform) (ovirtclient.Client, error) {
...
+	return nil
+}
+
+func readContentFromFileGlob(workDir, fileGlob string) ([]byte, error) {
+	globFullPath := filepath.Join(workDir, "openshift", fileGlob)
+	fullPaths, _ := filepath.Glob(globFullPath)
+
+	if fullPaths == nil {
+		return nil, errors.New(fmt.Sprintf("unable to find file '%s'", globFullPath))
+	}
```
Comment (osherdp) [inline internal/provider/ovirt/ignition.go:134]: I think it's worth propagating the error, for cases where we have a malformed pattern

File: internal/provider/ovirt/ignition.go [hunk 2]
```diff
@@ -77,43 +80,87 @@ func getOvirtClient(params *models.OvirtPlatform) (ovirtclient.Client, error) {
...
+func writeContentFromFileGlob(workDir, fileGlob string, content []byte) error {
+	globFullPath := filepath.Join(workDir, "openshift", fileGlob)
+	fullPaths, _ := filepath.Glob(globFullPath)
+
+	if fullPaths == nil {
+		return errors.New(fmt.Sprintf("unable to find file '%s'", globFullPath))
+	}
+
+	if len(fullPaths) > 1 {
+		return errors.New(fmt.Sprintf("more than one file found '%s'", globFullPath))
```
Comment (osherdp) [inline internal/provider/ovirt/ignition.go:157]:
```suggestion
		return errors.New(fmt.Sprintf("more than one file found for glob '%s': %v", globFullPath, fullPaths))
```

File: internal/provider/registry/registry_test.go [hunk 3]
```diff
@@ -400,6 +448,119 @@ var _ = Describe("Test SetPlatformUsages", func() {
...
+		Expect(err).To(BeNil())
+		vmName := vm.Name()
+		template, err := ovirtClient.GetTemplate(vm.TemplateID(), retryStragegy)
+		Expect(err).To(BeNil())
+		templateName := template.Name()
+		found := false
+		for i := 0; i < 3; i++ {
+			fileName := fmt.Sprintf(strings.Replace(ovirt.MachineManifestFileNameGlobStrFmt, "*", filePrefix, -1), i)
+			filePath := filepath.Join(baseDir, fileName)
+			fmt.Println(filePath)
```
Comment (osherdp) [inline internal/provider/registry/registry_test.go:550]: is it for debugging?

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by osherdp** [comment]
@mmartinv can you elaborate on how it was manually tested? thanks in advance

File: internal/provider/ovirt/ignition.go (hunk 2 above)
Comment (mmartinv) [reply internal/provider/ovirt/ignition.go]: Done

**Comment by osherdp** [comment]
@mmartinv ``edge-verify-generated-code`` still failing
can you please do ``skipper make generate``?

**Review by osherdp — APPROVED** [review]

**Comment by osherdp** [comment]
> @openshift-bot: This pull request references [Bugzilla bug 2100857](https://bugzilla.redhat.com/show_bug.cgi?id=2100857), which is invalid:
> 
> * expected the bug to target the "rhacm-2.6" release, but it targets "---" instead
> 
> Comment `/bugzilla refresh` to re-evaluate validity if changes to the Bugzilla bug are made, or edit the title of this pull request to link to a different bug.

Bug needs to be adjusted to be able to be linked to the PR. See other PRs for examples
/hold

**Pull request merged**